### PR TITLE
Fix flaky MultiOperation test

### PR DESCRIPTION
**Pull request by stephanos**
## What changed?

Fixing flakiness of MultiOperation test where workflow id conflict policy is "terminate".

### CODE REVIEW

File: tests/workflow.go [hunk 1]
```diff
@@ -1117,6 +1140,38 @@ func (s *FunctionalSuite) TestExecuteMultiOperation() {
...
+						}
+
+						_, err := poller.PollAndProcessWorkflowTask(WithDumpHistory)
+						var notFound *serviceerror.NotFound
+						if errors.As(err, &notFound) {
+							continue
+						}
+						s.NoError(err)
+					}
+				}()
```
Comment (stephanos) [inline tests/workflow.go:1173]: This is the fix for the flaky test.

File: tests/workflow.go [hunk 2]
```diff
@@ -1117,6 +1140,40 @@ func (s *FunctionalSuite) TestExecuteMultiOperation() {
...
+						if errors.As(err, &notFound) {
+							// If the Workflow is terminated *after* the WFT is picked up by the poller, but *before*
+							// it is completed, the History service responds with a "Workflow Task not found" error.
+							// In that case, we poll again until we completed the Update.
+							continue
+						}
+						s.NoError(err)
+					}
+				}()
+
```
Comment (stephanos) [inline tests/workflow.go:1176]: This fixes the flaky test.

File: tests/workflow.go [hunk 3]
```diff
@@ -986,42 +987,20 @@ func (s *FunctionalSuite) TestExecuteMultiOperation() {
...
+		reqDone := make(chan struct{})
 		go func() {
 			resp, retErr = s.engine.ExecuteMultiOperation(NewContext(), request)
-			done <- struct{}{}
+			reqDone <- struct{}{}
 		}()
 
-		_, err := poller.PollAndProcessWorkflowTask(WithDumpHistory)
-		s.NoError(err)
-
```
Comment (stephanos) [inline tests/workflow.go:1017]: I needed to customize the polling behavior for the particular testing scenario.

File: tests/workflow.go [hunk 4]
```diff
@@ -1061,6 +1060,7 @@ func (s *FunctionalSuite) TestExecuteMultiOperation() {
 				s.NotZero(startRes.RunId)
 
 				updateRes := resp.Responses[1].Response.(*workflowservice.ExecuteMultiOperationResponse_Response_UpdateWorkflow).UpdateWorkflow
+				s.NotNil(updateRes.Outcome)
```
Comment (stephanos) [inline tests/workflow.go:1063]: I noticed during testing that the assertion below isn't sufficient (in the Update timeout case)

**Review by alexshtin — APPROVED** [review]

**Comment by alexshtin** [comment]
Yohoo!

**Pull request merged**